### ISSUE TITLE
On MacOS M1 machines, the default renderer opens the Sketch window behind other windows

### ISSUE THREAD

**Issue by hx2A** [py5coding/py5generator#564]
As discussed in #562 , the following code opens the Sketch window behind the terminal window.

```python
import py5

def setup():
    py5.size(100,30)
    global x,y
    (x,y) = (0,15)
    
def draw():
    global x
    x = x + 1
    py5.ellipse(x,y,10,10)
    if x > 100 : py5.no_loop()

py5.run_sketch()
```

Workarounds are to use the `P2D` renderer or to call `py5.get_surface().set_always_on_top(True)`. This should work correctly though.

**Comment by hx2A** [comment]
> py5.get_surface().set_always_on_top(True) works fine, thanks. Using py5.size(100,100,py5.P2D) --> Python crash !! Mac mini M1

So, same code as above, but with the P2D renderer, in the generic python interpreter causes py5 to crash? I will look into this further later today.

**Comment by wissme** [comment]
Let me refine my sketch. After execution in IDLE 3.13 on a Mac mini M1 (MacOS 14.7), I can see "End!" so the sketch has terminated. The python shell is blocked (acceptable). Now I close the sketch window with the mouse click. The window closes, 'End' is NOT written at the Python shell, which his still blocked. I must restart the shell manually. I tried with Spyder 6.0.2 app on Mac, installed spyder-kernels=3.0, opened a console in py5 env and it's ok, the py5 env. console is not blocked even during the sketch (with 'End' coming immediately !). So the problem may be tied to IDLE 3.13 ? I use both IDLE and Spyder for a book in preparation...  Hum, I notice that if I do not close the sketch window and quit Spyder app (yeah bad), the Spyder app hangs, I must force kill it :-( The app is killed BUT the sketch window is still here, with the diabolic wheel turning, impossible to force quit. Gasp. Maybe test a bit more on Mac ARM machines ?...
```
%gui osx            # for Spyder 6.0.2 IPython

import py5         # 0.10.3a1

def setup():
    py5.size(100,100)
    py5.get_surface().set_always_on_top(True)    # ok but I hope this will be fixed :-)
    global x,y
    (x,y) = (0,50)
    
def draw():
    global x
    x = x + 1
    py5.ellipse(x,y,20,20)
    if x > 100 : py5.no_loop()

py5.run_sketch()
print('End')
```

**Comment by hx2A** [comment]
Hmmm, I just tried this example in IPython and the regular Python interpreter. For both cases, I was able to exit the Sketch window without any problems. For the regular Python interpreter I do not see the word "Exit" printed. This is because for py5 to run at all in the Python interpreter, it needs to do some complicated things that as a side effect guarantee that the Python interpreter will end when the Sketch ends. Python will never return from the `run_sketch()` call.

Getting py5 to run at all on MacOS computers has been by itself a huge challenge. It actually didn't work at all for the first year of py5's existence. (A big factor here is that this was during the pandemic and I did not yet have a MacOS computer to test on or do development on.) It does work now, but as you can see, there are some areas where it is not perfect. Note that Processing itself can side-step the issues because when a Processing Sketch exits, it will call `System.exit()`, killing the JVM. py5 does not have that luxury so I needed to figure out a way to exit the Sketch for all renderers and on all operating systems. This was not an easy task!

**Comment by hx2A** [comment]
OK, I just tested the following code on a MacOS M1 machine and  MacOS machine with an intel Processor:

```python
import py5

def setup():
    py5.size(200, 200)

def draw():
    py5.rect(py5.mouse_x, py5.mouse_y, 10, 10)

py5.run_sketch()
```

On the M1 machine, the window opened behind the terminal window. On the Intel machine, it opened in front of the terminal window. It seems there is an unfortunate difference between the two.

Also, this bug is closely related to #5. I created a fix that felt like a hack so so left that bug open. Let me take a closer look at this and see what I can do to improve things.
